marimo/_ast/cell_manager.py

```diff
@@ -386,10 +386,13 @@ def sort_cell_ids_by_similarity(
         id_mapping = dict(zip(sorted_ids, current_ids))
 
         # Update the cell data in place
-        self._cell_data = {
-            new_id: self._cell_data[old_id]
-            for new_id, old_id in id_mapping.items()
-        }
+        new_cell_data: dict[CellId_t, CellData] = {}
+        for new_id, old_id in id_mapping.items():
+            prev_cell_data = self._cell_data[old_id]
+            prev_cell_data.cell_id = new_id
```

Comment:
`prev_cell_data.cell_id = new_id` was the incorrect bit (b/c there is 2 sources of truth for the ids)